mockgen/model/model.go

```diff
@@ -415,10 +415,31 @@ func typeFromType(t reflect.Type) (Type, error) {
 	}
 
 	if imp := t.PkgPath(); imp != "" {
-		return &NamedType{
+
+		typeObj := &NamedType{
 			Package: impPath(imp),
 			Type:    t.Name(),
-		}, nil
+		}
+		if typeName := t.Name(); strings.ContainsAny(typeName, "[]") {
+			var ts []Type
+			startIndex := strings.Index(typeName, "[")
+			endIndex := strings.Index(typeName, "]")
```

Comment:
Golang's type parameter can be a slice `X[]` or another type parameter, therefore, I am afraid the assumption here for unwrapping `[]` is incorrect giving case like `foo.bar[baz[]]`.

FWIW, since golang's reflection doesn't provide the type parameter details, we don't have a good way to tell what is the concrete type used. One way I was attempting to use is to use a recursive parser to extract the nested type parameters, but not that simple to write right... Another way is to wait for support from `reflect`...

Reply:
I'll try the recursive parser. Thanks for the hint!